p2p/switch.go

```diff
@@ -341,6 +341,11 @@ func (sw *Switch) SetAddrBook(addrBook AddrBook) {
 	sw.addrBook = addrBook
 }
 
+// SetPeerConfig sets the PeerConfig used for peer setup.
+func (sw *Switch) SetPeerConfig(config *PeerConfig) {
```

Comment:
maybe while you're working on this you can try to tackle https://github.com/tendermint/tendermint/blob/8393d4717fb1cfff28a9eaff2407165cdae7d9a3/p2p/switch.go#L93

would be good to avoid having this setter (SetPeerConfig).

Reply:
@melekes I'd like to do this in a separate follow-up PR, can we approve this one for now?

Reply:
ok, sure